mooncake-store/src/master.cpp

```diff
@@ -69,45 +71,45 @@ DEFINE_string(cluster_id, mooncake::DEFAULT_CLUSTER_ID,
+    default_config.GetUInt64("default_kv_soft_pin_ttl", &master_config.default_kv_soft_pin_ttl, mooncake::DEFAULT_KV_SOFT_PIN_TTL_MS);
+    default_config.GetBool("allow_evict_soft_pinned_objects",
+                            &master_config.allow_evict_soft_pinned_objects, mooncake::DEFAULT_ALLOW_EVICT_SOFT_PINNED_OBJECTS);
+    default_config.GetDouble("eviction_ratio", &master_config.eviction_ratio, mooncake::DEFAULT_EVICTION_RATIO);
+    default_config.GetDouble("eviction_high_watermark_ratio",
+                                &master_config.eviction_high_watermark_ratio, mooncake::DEFAULT_EVICTION_HIGH_WATERMARK_RATIO);
+    default_config.GetInt64("client_live_ttl_sec", &master_config.client_live_ttl_sec, mooncake::DEFAULT_CLIENT_LIVE_TTL_SEC);
 
-int main(int argc, char* argv[]) {
-    easylog::set_min_severity(easylog::Severity::WARN);
-    // Initialize gflags
-    gflags::ParseCommandLineFlags(&argc, &argv, true);
-
-    LOG(INFO) << "Master service started on port " << FLAGS_port
-              << ", enable_gc=" << FLAGS_enable_gc
-              << ", max_threads=" << FLAGS_max_threads
-              << ", enable_metric_reporting=" << FLAGS_enable_metric_reporting
-              << ", metrics_port=" << FLAGS_metrics_port
-              << ", default_kv_lease_ttl=" << FLAGS_default_kv_lease_ttl
-              << ", default_kv_soft_pin_ttl=" << FLAGS_default_kv_soft_pin_ttl
-              << ", allow_evict_soft_pinned_objects="
-              << FLAGS_allow_evict_soft_pinned_objects
-              << ", eviction_ratio=" << FLAGS_eviction_ratio
-              << ", eviction_high_watermark_ratio="
-              << FLAGS_eviction_high_watermark_ratio
-              << ", enable_ha=" << FLAGS_enable_ha
-              << ", etcd_endpoints=" << FLAGS_etcd_endpoints
-              << ", client_ttl=" << FLAGS_client_ttl
-              << ", rpc_thread_num=" << FLAGS_rpc_thread_num
-              << ", rpc_port=" << FLAGS_rpc_port
-              << ", rpc_address=" << FLAGS_rpc_address
-              << ", rpc_conn_timeout_seconds=" << FLAGS_rpc_conn_timeout_seconds
-              << ", rpc_enable_tcp_no_delay=" << FLAGS_rpc_enable_tcp_no_delay
-              << ", cluster_id=" << FLAGS_cluster_id
-              << ", memory_allocator=" << FLAGS_memory_allocator;
-
-    int server_thread_num =
-        std::min(FLAGS_max_threads,
-                 static_cast<int>(std::thread::hardware_concurrency()));
+    default_config.GetBool("enable_ha", &master_config.enable_ha, false);
+    default_config.GetString("etcd_endpoints", &master_config.etcd_endpoints, "");
+    default_config.GetString("cluster_id", &master_config.cluster_id, mooncake::DEFAULT_CLUSTER_ID);
+    default_config.GetString("memory_allocator", &master_config.memory_allocator, "offset");
+}
 
+void LoadConfigFromCmdline(mooncake::MasterConfig& master_config) {
     if (FLAGS_max_threads != 4) {  // 4 is the default value
```

Comment:
These codes need to be executed after the final values of the input parameters are determined, i.e., after LoadConfigFromCmdline.